src/utils/component-swap-data.ts

```diff
@@ -196,49 +198,28 @@ async function getAmount(
     const publicClient = createClient(chainId)!
     const erc4626Abi = [
       'function convertToAssets(uint256 shares) view returns (uint256 assets)',
-      'function previewDeposit(uint256 assets) view returns (uint256)',
-      'function previewMint(uint256 shares) view returns (uint256)',
-      'function previewRedeem(uint256 shares) view returns (uint256)',
-      'function previewWithdraw(uint256 assets) view returns (uint256)',
+      'function previewMint(uint256 shares) view returns (uint256 assets)',
+      'function previewRedeem(uint256 shares) view returns (uint256 assets)',
     ]
-    const assets: bigint = (await publicClient.readContract({
+    const preview: bigint = (await publicClient.readContract({
       address: component as Address,
-      abi: erc4626Abi,
-      functionName: 'convertToAssets',
+      abi: parseAbi(erc4626Abi),
+      functionName: isMinting ? 'previewMint' : 'previewRedeem',
       args: [issuanceUnits],
     })) as bigint
-    return assets
+    return preview
   } catch {
-    // TODO: apply slippage to issuance units amount (for all none erc4262)
-    return issuanceUnits
+    // Apply slippage to issuance units amount (for all none erc4262)
+    if (isMinting) {
+      return (issuanceUnits * BigInt(1005)) / BigInt(1000)
+    } else {
+      return (issuanceUnits * BigInt(995)) / BigInt(1000)
+    }
```

Comment:
@pblivin0x @edkim went with 0.5% slippage for now. Let me know if you think that's a good value.